pkg/controller/fileobserver/observer.go

```diff
@@ -40,20 +40,26 @@ func (t ActionType) String(filename string) string {
 	return ""
 }
 
-// reactorFn define a reaction function called when an observed file is modified.
-type reactorFn func(file string, action ActionType) error
+// ReactorFn define a reaction function called when an observed file is modified.
+type ReactorFn func(file string, action ActionType) error
 
 // ExitOnChangeReactor provides reactor function that causes the process to exit when the change is detected.
-var ExitOnChangeReactor reactorFn = func(filename string, action ActionType) error {
-	klog.Infof("exiting because %q changed", filename)
-	os.Exit(0)
-	return nil
+// DEPRECATED: Using this function cause process to exit immediately without proper shutdown (context close/etc.)
+//             Use the TerminateOnChangeReactor() instead.
+var ExitOnChangeReactor = TerminateOnChangeReactor(func() { os.Exit(0) })
+
+func TerminateOnChangeReactor(terminateFn func()) ReactorFn {
```

Comment:
a builder :D I was joking ;-)

Reply:
i like it better now ;-) even if it was just a joke